includes/class-wc-gateway-ppec-client.php

```diff
@@ -1012,7 +1012,37 @@ public function get_do_reference_transaction_params( array $args ) {
 			) ),
 		);
 
-		if ( ! empty( $details['shipping_address'] ) ) {
+		if ( ! empty( $details['shipping_address'] ) &&
```

Comment:
I found this a bit hard to get my head around. Would creating a new method help, or would that confuse things more?

```
if ( ! empty( $details['shipping_address'] ) && $details['shipping_address']->is_complete_shipping_address() ) {
```

Maybe on the address object, or if it feels too specific to `get_do_reference_transaction_params` then it could just be a seperate function in the `WC_Gateway_PPEC_Client` class?

Reply:
Thanks for pointing it out. Wasn't sure off the bat whether it was unreadable.

Added it to the address class - turns out many different API calls have the same shipping requirements so I didn't have to call it `has_all_required_shipping_params_for_do_reference_transaction` 😄